src/connectedk8s/TroubleshootTemplates/diagnoser_job.yaml

```diff
@@ -0,0 +1,52 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: azure-arc-troubleshoot
+  namespace: default
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  namespace: default
+  name: secret-reader
+rules:
+- apiGroups: [""] # "" indicates the core API group
+  resources: ["secrets"]
+  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
```

Comment:
We don't need access to delete/update/patch secrets. We just need access to list them, test it out with just, "get", "list" in this section and remove rest of the ones